dpc-api/src/main/java/gov/cms/dpc/api/resources/v1/PatientResource.java

```diff
@@ -336,4 +344,46 @@ private static Optional<WebApplicationException> validateAndAddOrg(Patient patie
             return Optional.empty();
         }
     }
+
+    private IQuery<Bundle> buildPatientSearchQuery(String orgId, @Nullable String patientMBI) {
+        IQuery<Bundle> query = this.client
+                .search()
+                .forResource(Patient.class)
+                .encodedJson()
+                .where(Patient.ORGANIZATION.hasId(orgId))
+                .returnBundle(Bundle.class);
+        if (patientMBI != null && !patientMBI.isEmpty()) {
+            final String expandedMBI;
+            if (IDENTIFIER_PATTERN.matcher(patientMBI).matches()) {
+                expandedMBI = patientMBI;
+            } else {
+                expandedMBI = String.format("%s|%s", DPCIdentifierSystem.MBI.getSystem(), patientMBI);
+            }
+            query = query.where(Patient.IDENTIFIER.exactly().identifier(expandedMBI));
+        }
+
+        return query;
+    }
+
+    private void handlePagingLinkPathing(Bundle bundle) {
+        if (bundle == null || bundle.getLink() == null) {
```

Comment:
```suggestion
        if (bundle == null || bundle.hasLink() {
```

HAPI has a bunch of methods like this you can use to simplify null and empty checks.

Reply:
updated to use `!bundle.hasLink()`